### PR TITLE
PyCaret Integration

### LINKED ISSUE #8167: [Integration] Integrate PyCaret with MindsD
The goal of this issue is to build a new handler that will integrate PyCaret into MindsDB.

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Building ML Handler docs](https://docs.mindsdb.com/contribute/ml-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices.

* :book:  [PyCaret](https://pycaret.org/)

* :point_right: [LlamaIndex Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/llama_index_handler): This existing handler connects MindsDB with LlamaIndex.

## Next Steps :male_detective: :female_detective: 
 
-   You can familiarize yourself with PyCaret.
-   Create a new handler that connects MindsDB to PyCaret.
-   Implement the core features mentioned above.
-   Thoroughly test the integration to ensure proper functionality.
-   Submit a pull request with the new PyCaret Integration handler.

**The PR should include the implementation of the handler and unit tests.**

## :loudspeaker: Additional rewards :trophy:

Each ML integration brings :seven: :zero:  : points for SWAG and entry into the draw for a  :point_down: :
- :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by aditya-azad**
## Description

Integrates PyCaret with MindsDB. The only issue is that when tests are run individually, they work, but I get "Prediction target not found" error when running all the pycaret tests at once.

Fixes #8167 

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

Screenshot 2023-10-28 180918

Screenshot 2023-10-28 180927

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Review by ZoranPandovski — DISMISSED** [review]
Can we reference the tests files from some location instead of adding them to the codebase?

**Comment by aditya-azad** [comment]
> Can we reference the tests files from some location instead of adding them to the codebase?

What do you mean by that?

**Comment by ZoranPandovski** [comment]
The csv files, we don't want to keep them in the handler

File: tests/unit/ml_handlers/test_pycaret.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        if ret.data is not None:
+            columns = [
+                col.alias if col.alias is not None else col.name
+                for col in ret.columns
+            ]
+            return pd.DataFrame(ret.data, columns=columns)
+
+    @patch('mindsdb.integrations.handlers.postgres_handler.Handler')
+    def test_classifier(self, mock_handler):
+        iris_df = pd.read_csv('https://raw.githubusercontent.com/pycaret/pycaret/master/datasets/iris.csv')
```
Comment (dusvyat) [inline tests/unit/ml_handlers/test_pycaret.py:50]: @aditya-azad - seems a bit unnecessary to download actual real data for unit tests. For speed and realiability reasons mostly.

I recon some fake/mock data should do the trick.

**Review by dusvyat — CHANGES_REQUESTED** [review]

File: tests/unit/ml_handlers/test_pycaret.py (hunk 1 above)
Comment (aditya-azad) [reply tests/unit/ml_handlers/test_pycaret.py]: Sure thing!

**Review by dusvyat — APPROVED** [review]
@aditya-azad - lgtm - one minor point, as Zoran noted above please move the tests into 'test' sub directory within pycaret_handler folder. 

After that, I will go ahead and merge

**Review by dusvyat — APPROVED** [review]
thanks for your patience, will merge now

**Pull request merged**